File: backend/app/models/performance.py
"""策略绩效追踪数据模型。"""

from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel


class SettlementStatus(str, Enum):
    """策略结算状态。"""

    PENDING = "pending"
    HIT_STOP_LOSS = "hit_stop_loss"
    HIT_TARGET = "hit_target"
    TIMEOUT = "timeout"


class StrategyDirection(str, Enum):
    """策略方向。"""

    LONG = "long"
    SHORT = "short"
    NEUTRAL = "neutral"


class StrategySnapshotCreate(BaseModel):
    """创建策略快照的请求模型。"""

    strategy_id: UUID
    symbol: str
    direction: StrategyDirection
    entry_low: float
    entry_high: float
    stop_loss: float
    targets: list[float]
    confidence: float
    price_at_generation: float
    user_id: UUID | None = None
    analysis_mode: str | None = None


class PerfCheckpoint(BaseModel):
    """策略生成后的定时价格记录点。"""

    snapshot_id: UUID
    checkpoint_hours: int
    actual_price: float
    recorded_at: datetime


class SettlementResult(BaseModel):
    """策略结算结果。"""

    snapshot_id: UUID
    status: SettlementStatus
    settlement_price: float
    settlement_time: datetime
    pnl_pct: float


class PerformanceStats(BaseModel):
    """绩效统计汇总。"""

    total_strategies: int
    settled_count: int
    win_rate: float
    avg_profit_pct: float
    avg_loss_pct: float
    profit_loss_ratio: float
    sharpe_ratio: float = 0.0
    by_agent: dict[str, float]

File: frontend/lib/api/performance.ts
import { authFetch } from "./auth";
import { handleApiResponse } from "./helpers";

const API_BASE = process.env.NEXT_PUBLIC_API_URL || "";

// ── Types ────────────────────────────────────────────────────

export interface PerformanceStats {
  total_strategies: number;
  settled_count: number;
  win_rate: number;
  avg_profit_pct: number;
  avg_loss_pct: number;
  profit_loss_ratio: number;
  sharpe_ratio: number;
  by_agent: Record<string, number>;
}

export interface SnapshotCheckpoint {
  snapshot_id: string;
  checkpoint_hours: number;
  actual_price: number;
  recorded_at: string;
}

export interface SnapshotDetail {
  snapshot: {
    id: string;
    strategy_id: string;
    symbol: string;
    direction: string;
    entry_low: number;
    entry_high: number;
    stop_loss: number;
    targets: number[];
    confidence: number;
    price_at_generation: number;
    status: string;
    settlement_price: number | null;
    settlement_time: string | null;
    pnl_pct: number | null;
    created_at: string;
  };
  checkpoints: SnapshotCheckpoint[];
}

export interface TrendDataPoint {
  date: string;
  win_rate: number;
  cumulative_pnl: number;
}

// ── API calls ────────────────────────────────────────────────

export async function getStats(
  symbol?: string,
  days: number = 30,
  direction?: string
): Promise<PerformanceStats> {
  const params = new URLSearchParams();
  if (symbol) params.set("symbol", symbol);
  params.set("days", String(days));
  if (direction) params.set("direction", direction);

  const res = await authFetch(
    `${API_BASE}/api/performance/stats?${params.toString()}`
  );
  return handleApiResponse(res, "获取绩效统计失败");
}

export async function getSnapshotDetail(
  snapshotId: string
): Promise<SnapshotDetail> {
  const res = await authFetch(
    `${API_BASE}/api/performance/snapshots/${encodeURIComponent(snapshotId)}`
  );
  return handleApiResponse(res, "获取快照详情失败");
}

export async function getTrend(
  days: number = 30
): Promise<TrendDataPoint[]> {
  const res = await authFetch(
    `${API_BASE}/api/performance/trend?days=${days}`
  );
  return handleApiResponse(res, "获取趋势数据失败");
}

export const performanceApi = {
  getStats,
  getSnapshotDetail,
  getTrend,
};
